pkg/reconciler/v1alpha1/autoscaling/autoscaling.go

```diff
@@ -247,15 +246,20 @@ func (c *Reconciler) updateStatus(desired *kpa.PodAutoscaler) (*kpa.PodAutoscale
 	if err != nil {
 		return nil, err
 	}
-	// Check if there is anything to update.
-	if !reflect.DeepEqual(kpa.Status, desired.Status) {
-		// Don't modify the informers copy
-		existing := kpa.DeepCopy()
-		existing.Status = desired.Status
-
-		// TODO: for CRD there's no updatestatus, so use normal update
-		return c.ServingClientSet.AutoscalingV1alpha1().PodAutoscalers(kpa.Namespace).Update(existing)
-		//	return prClient.UpdateStatus(newKPA)
+	// If there's nothing to update, just return.
+	if reflect.DeepEqual(kpa.Status, desired.Status) {
+		return kpa, nil
 	}
-	return kpa, nil
+	// Don't modify the informers copy
+	existing := kpa.DeepCopy()
+	existing.Status = desired.Status
+
+	// TODO: for CRD there's no updatestatus, so use normal update
+	updated, err := c.ServingClientSet.AutoscalingV1alpha1().PodAutoscalers(kpa.Namespace).Update(existing)
+	if err != nil {
+		return nil, err
+	}
+
+	c.Recorder.Eventf(desired, corev1.EventTypeNormal, "Updated", "Updated status for KPA %q", desired.Name)
```

Comment:
I'm not convinced of the usefulness of this event...  In some cases we update status a fair amount before we're done, which will mean a lot of events.

Reply:
For KPA specifically or for our CRDs in general?

Reply:
I'm also concerned that we may get flooded by unrelated events.
But on the other hand, I feel it acceptable to add first, and do subtraction later if we indeed find annoying events later.

Reply:
Generally I don't find it useful to surface an event that just says: "I updated status!" without what that update is.

I feel like this could be made useful by surfacing events reflecting changes in our conditions (vs. existing), especially with https://github.com/knative/serving/pull/2436 giving a hit for the level of event we should be surfacing (on False).

I'd exclude this from all resource types for now, and we can open an issue (for followup) that takes:
`(r Recorder, obj runtime.Object, before duckv1alpha1.Conditions, after duckv1alpha1.Conditions) -> error`

We should be able to take a generic delta of the `Conditions` and surface events through the `Recorder` for transitions.